test/lib/jdk/test/lib/process/ProcessTools.java

```diff
@@ -430,6 +455,41 @@ public static OutputAnalyzer executeProcess(ProcessBuilder pb, String input) thr
      */
     public static OutputAnalyzer executeProcess(ProcessBuilder pb, String input,
                                                 Charset cs) throws Exception {
+        return executeProcessPreservePID(pb, input, cs).output();
+    }
+
+    static public class OutputAnalyzerAndPID {
```

Comment:
can we either change `OutputAnalyzer` to store pid (and use -1 for cases when there is no one) or make `OutputAnalyzer` non-final and have `OutputAnalyzerAndPID` extending `OutputAnalyzer`?

Reply:
Thanks for reviewing this. I did not store the pid in the OutputAnalyzer because it doesn't seem to belong there as it has nothing to do with the text output of a test. But if you think that's ok. that's fine with me too. Do you prefer an extra field in OutputAnalyzer or a new subclass?

Reply:
`OutputAnalyzer`/ `OutputBuffer` already have an exit value, which arguable has nothing to do w/ the text output either, so I think it's ok to add a pid into `OutputBuffer` i-face and both of its implementations.

Reply:
I just pushed a change that does this. Does that look ok to you?

Reply:
yes, thank you.